optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -54,14 +54,25 @@ def __init__(
 
     def compile(self):
         if self.request is None:
-            logger.info(f"Compiling the Language model to {self._device} ...")
-            self.request = core.compile_model(self.model, self._device, self.ov_config).create_infer_request()
+            if self._compile_only:
+                self.request = self.model.create_infer_request()
+            else:
+                logger.info(f"Compiling the Language model to {self._device} ...")
+                self.request = self._compile_model(
+                    self.model, self._device, self.ov_config, self.model_save_dir
+                ).create_infer_request()
```

Comment:
why can't we call https://github.com/huggingface/optimum-intel/blob/cead516b3b8c38c1a0b8b8937a861edccdfab839/optimum/intel/openvino/modeling_decoder.py#L396-L401 here ?

Reply:
not related to this PR but I find having `compile_model` / `compile` / `_compile` / `_compile_text_emb` makes it super difficult to follow everything that we are doing here, not sure how we can make this clearer

Reply:
for the first comment, I reused compile for decoder, for the second one, I do not know what I can suggest for now. The problem that we have 2 requests in model (for llm part and separated for embeddings. Adding one more level and separated model may be solution for having own compilation method, but it is also I think may make model usage complicated as there is no possibility to use models separately